realm/realm-library/src/objectServer/java/io/realm/SyncSession.java

```diff
@@ -387,7 +388,12 @@ private void waitForChanges(int direction) throws InterruptedException {
 
                 throw new ObjectServerError(ErrorCode.UNKNOWN, errorMsg + " Has the SyncClient been started?");
             }
-            wrapper.waitForServerChanges();
+            try {
+                wrapper.waitForServerChanges();
+            } catch(InterruptedException e) {
+                waitingForServerChanges.set(null); // Ignore any results being sent if the wait was interrupted.
```

Comment:
Hmm, good point. It looks like this will require changes to object store as it stands though. Right now we don't have any way to cancel this by the looks of it